docs/source/concepts.rst

```diff
@@ -352,6 +352,71 @@ models are depending on each other, such as the example above.
 
 .. _concepts-api-func-and-adapters:
 
+Model Artifact Metadata
+^^^^^^^^^^^^^^^^^^^^^^^
+
+BentoML allows users to include additional metadata information for the packed model
+artifacts. The metadata are intended to be used to specify identifying attributes of
+the model artifact that are meaningful and relevant to users.
+
+Users can add metadata information, along with the model artifact.
+
+.. code-block:: python
+
+    # Using the example above.
+    svc = MyPredictionService()
+    svc.pack(
+        'model_a',
+        my_sklearn_model_object,
+        metadata={
+            'precision_score': 0.876,
+            'created_by': 'joe'
+        }
+    )
+    svc.pack(
+        'model_b',
+        my_xgboost_model_object,
+        metadata={
+            'precision_score': 0.792,
+            'mean_absolute_error': 0.88
+        }
+    )
+    svc.save()
+
+.. note::
+
+    The model artifact metadata is immutable.
+
```

Comment:
might be more clear to put this above the code block

Reply:
Yeah, might be.  I am trying to catch readers' attention on this.  Maybe have the message in the code is more natural.